### PR TITLE
Add the feature of comparing variables to reference files

### PR DESCRIPTION
**Pull request by apchoiCMD**
#### Work added
- add the `testoutput` method into base class
- now be able to see a basic stat within `ctest`

### CODE REVIEW

**Comment by guillaumevernieres** [comment]
@apchoiCMD , I thought we decided last week to move the iodavars.testoutput call in NetCDFToIodaConverter.h . That didn't work?

**Comment by apchoiCMD** [comment]
> @apchoiCMD , I thought we decided last week to move the iodavars.testoutput call in NetCDFToIodaConverter.h . That didn't work?

@guillaumevernieres Yes, we did but it didn't work for me and ```util.h``` has well organized functions so that I decided to use as same as ghrsst2ioda.h , but we will need to discuss it.

**Comment by guillaumevernieres** [comment]
> > @apchoiCMD , I thought we decided last week to move the iodavars.testoutput call in NetCDFToIodaConverter.h . That didn't work?
> 
> @guillaumevernieres Yes, we did but it didn't work for me and `util.h` has well organized functions so that I decided to use as same as ghrsst2ioda.h , but we will need to discuss it.

I don't know why you feel the need to add a laugh emoji to all my posts @apchoiCMD . I was not trying to be funny with the question above. 
What exactly is it that you tried that did not work? I don't understand what you mean with the 2 bullets under **Work needed if possible**. I don't mind merging as is if there's an issue moving the `testoutput` to the base class, but I'd need to understand what the issue is first.

**Comment by apchoiCMD** [comment]
> > > @apchoiCMD , I thought we decided last week to move the iodavars.testoutput call in NetCDFToIodaConverter.h . That didn't work?
> > 
> > 
> > @guillaumevernieres Yes, we did but it didn't work for me and `util.h` has well organized functions so that I decided to use as same as ghrsst2ioda.h , but we will need to discuss it.
> 
> I don't know why you feel the need to add a laugh emoji to all my posts @apchoiCMD . I was not trying to be funny with the question above. What exactly is it that you tried that did not work? I don't understand what you mean with the 2 bullets under **Work needed if possible**. I don't mind merging as is if there's an issue moving the `testoutput` to the base class, but I'd need to understand what the issue is first.

@guillaumevernieres here is my thought, `util.h` has now all functions that connected to all converters as well as NetCDFToIodaConverter.h and I thought you separated it because you wanted to make NetCDFToIodaConverter.h as simple as possible. As I saw functions that are append, trim and output in `util.h` together, I don't think it makes a sense that we move `output` only. Correct me if I am wrong and removed **Work needed if possible**

File: utils/obsproc/IcecAmsr2Ioda.h [hunk 1]
```diff
@@ -94,6 +95,9 @@ namespace gdasapp {
         && iodaVars.datetime_ > 0.0);
       iodaVars.trim(mask);
 
+      // Test output
```
Comment (guillaumevernieres) [inline utils/obsproc/IcecAmsr2Ioda.h]: Move this to the base-class. I just tested it and it works fine, you just need to adjust your reference file.

File: utils/obsproc/Rads2Ioda.h [hunk 2]
```diff
@@ -135,6 +136,9 @@ namespace gdasapp {
           }
       }
 
+       // Test output
```
Comment (guillaumevernieres) [inline utils/obsproc/Rads2Ioda.h]: same comment as above

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
Looks good @apchoiCMD , my only request is to move the call to iodavars.testoutput to the base class. I just tested it and it works fine.

**Review by guillaumevernieres — APPROVED** [review]
Looks good. Thanks @apchoiCMD .

**Pull request merged**